File: backend/src/main/java/com/contentplatform/backend/infrastructure/jpa/entity/SitemapTemplateEntity.java
package com.contentplatform.backend.infrastructure.jpa.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.Instant;

@Entity
@Table(name = "sitemap_templates")
public class SitemapTemplateEntity {
    @Id
    @Column(name = "id", nullable = false, length = 36)
    private String id;

    @Column(name = "tenant_id", nullable = false, length = 36)
    private String tenantId;

    @Column(name = "content_type", nullable = false, length = 64)
    private String contentType;

    @Column(name = "enabled", nullable = false)
    private boolean enabled;

    @Column(name = "template", length = 512)
    private String template;

    @Column(name = "lastmod_policy", nullable = false, length = 32)
    private String lastmodPolicy;

    @Column(name = "default_changefreq", length = 16)
    private String defaultChangefreq;

    @Column(name = "default_priority", precision = 2, scale = 1)
    private Double defaultPriority;

    @Column(name = "validate_status", nullable = false, length = 16)
    private String validateStatus;

    @Column(name = "validate_errors", columnDefinition = "jsonb")
    private String validateErrors;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    protected SitemapTemplateEntity() {
    }

    public SitemapTemplateEntity(String id,
                                 String tenantId,
                                 String contentType,
                                 boolean enabled,
                                 String template,
                                 String lastmodPolicy,
                                 String defaultChangefreq,
                                 Double defaultPriority,
                                 String validateStatus,
                                 String validateErrors,
                                 Instant createdAt,
                                 Instant updatedAt) {
        this.id = id;
        this.tenantId = tenantId;
        this.contentType = contentType;
        this.enabled = enabled;
        this.template = template;
        this.lastmodPolicy = lastmodPolicy;
        this.defaultChangefreq = defaultChangefreq;
        this.defaultPriority = defaultPriority;
        this.validateStatus = validateStatus;
        this.validateErrors = validateErrors;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public String getId() {
        return id;
    }

    public String getTenantId() {
        return tenantId;
    }

    public String getContentType() {
        return contentType;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public String getTemplate() {
        return template;
    }

    public String getLastmodPolicy() {
        return lastmodPolicy;
    }

    public String getDefaultChangefreq() {
        return defaultChangefreq;
    }

    public Double getDefaultPriority() {
        return defaultPriority;
    }

    public String getValidateStatus() {
        return validateStatus;
    }

    public String getValidateErrors() {
        return validateErrors;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }
}

File: backend-nestjs/src/entities/sitemap-template.entity.ts
import { Column, CreateDateColumn, Entity, Index, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm';

export enum SitemapLastmodPolicy {
  UPDATED_AT = 'updatedAt',
  PUBLISHED_AT = 'publishedAt',
}

export enum SitemapValidateStatus {
  OK = 'OK',
  ERROR = 'ERROR',
  WARNING = 'WARNING',
}

@Entity({ name: 'sitemap_templates' })
@Index(['tenantId', 'contentType'], { unique: true })
export class SitemapTemplateEntity {
  @PrimaryGeneratedColumn('uuid')
  id!: string;

  @Column({ name: 'tenant_id', type: 'varchar', length: 36 })
  tenantId!: string;

  @Column({ name: 'content_type', type: 'varchar', length: 64 })
  contentType!: string;

  @Column({ type: 'boolean', default: false })
  enabled!: boolean;

  @Column({ type: 'varchar', length: 512, nullable: true })
  template!: string | null;

  @Column({
    name: 'lastmod_policy',
    type: 'enum',
    enum: SitemapLastmodPolicy,
    default: SitemapLastmodPolicy.UPDATED_AT,
  })
  lastmodPolicy!: SitemapLastmodPolicy;

  @Column({ name: 'default_changefreq', type: 'varchar', length: 16, nullable: true })
  defaultChangefreq!: string | null;

  @Column({
    name: 'default_priority',
    type: 'decimal',
    precision: 2,
    scale: 1,
    nullable: true,
  })
  defaultPriority!: string | null;

  @Column({
    name: 'validate_status',
    type: 'enum',
    enum: SitemapValidateStatus,
    default: SitemapValidateStatus.OK,
  })
  validateStatus!: SitemapValidateStatus;

  @Column({ name: 'validate_errors', type: 'jsonb', nullable: true })
  validateErrors!: string[] | null;

  @CreateDateColumn({ name: 'created_at', type: 'timestamptz' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'updated_at', type: 'timestamptz' })
  updatedAt!: Date;
}

